MonoGame.Framework/Graphics/TextureCollection.DirectX.cs

```diff
@@ -12,8 +12,14 @@ void PlatformInit()
 
         internal void ClearTargets(GraphicsDevice device, RenderTargetBinding[] targets)
         {
-            ClearTargets(targets, device._d3dContext.VertexShader);
-            ClearTargets(targets, device._d3dContext.PixelShader);
+            if (_applyToVertexStage)
+            {
+                ClearTargets(targets, device._d3dContext.VertexShader);
+            }
+            else
+            {
+                ClearTargets(targets, device._d3dContext.PixelShader);
+            }
```

Comment:
@tgjones - I wonder if we're better off just passing the DX pixel shader or vertex shader context on construction of the TextureCollection.  At least that cuts out some of the extra branches at runtime.